check_api/src/main/java/com/google/errorprone/BaseErrorProneJavaCompiler.java

```diff
@@ -199,24 +201,31 @@ static ErrorProneAnalyzer createAnalyzer(
     }
     refactoringCollection[0] = RefactoringCollection.refactor(epOptions.patchingOptions());
 
-    // Refaster refactorer or using builtin checks
-    CodeTransformer codeTransformer =
-        epOptions
-            .patchingOptions()
-            .customRefactorer()
-            .or(
-                () -> {
-                  ScannerSupplier toUse = ErrorPronePlugins.loadPlugins(scannerSupplier, context);
-                  Set<String> namedCheckers = epOptions.patchingOptions().namedCheckers();
-                  if (!namedCheckers.isEmpty()) {
-                    toUse = toUse.filter(bci -> namedCheckers.contains(bci.canonicalName()));
-                  }
-                  return ErrorProneScannerTransformer.create(toUse.applyOverrides(epOptions).get());
-                })
-            .get();
+    // Use Refaster refactorer if configured
+    Optional<Supplier<CodeTransformer>> customRefactorer =
+        epOptions.patchingOptions().customRefactorer();
+    CodeTransformer builtinTransformer =
+        createBuiltinRefactorer(scannerSupplier, epOptions, context, customRefactorer.isPresent());
+    CodeTransformer compoundTransformer =
+        customRefactorer
+            .transform(s -> CompositeCodeTransformer.compose(s.get(), builtinTransformer))
+            .or(builtinTransformer);
 
     return ErrorProneAnalyzer.createWithCustomDescriptionListener(
-        codeTransformer, epOptions, context, refactoringCollection[0]);
+        compoundTransformer, epOptions, context, refactoringCollection[0]);
+  }
+
+  private static CodeTransformer createBuiltinRefactorer(
+      ScannerSupplier scannerSupplier,
+      ErrorProneOptions epOptions,
+      Context context,
+      boolean forceFilter) {
```

Comment:
This `forceFilter` flag makes sure that the default patch mode of applying all built-in checks is disabled when a Refaster rule is supplied. I added this for backwards compatibility, but it does mean there's (still) no way to apply all built-in checks _and_ any additional Refaster rules. That's another way in which Refaster is "special" (contrast this with service-loaded custom plugins:  those are applied by default).

This is another thing I'd like to improve; input welcome!